File: mario6/mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break

    for i in range(height, 0, -1):
        printRow(i-1, height-(i-1))


def printRow(dot, hash):
    for i in range(dot):
        print(" ", end="")

    for j in range(hash):
        print("#", end="")

    print()


main()

File: probleme_rez/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int dot, int hash);

int main(void)
{
    int height = get_int("What is the height?");

    while(height <= 0)
    {
        height = get_int("What is the height?");
    }

    for(int i = height; i > 0; i--)
        {
             print_row(i-1, height-(i-1));
         }
}

void print_row(int dot, int hash)
{
    for(int i = 0; i < dot; i++)
    {
        printf(" ");
    }
    for(int j = 0; j < hash; j++)
    {
        printf("#");
    }
    printf("\n");
}
